cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ArtifactHttpHandler.java

```diff
@@ -342,11 +342,18 @@ public void writeProperties(FullHttpRequest request, HttpResponder responder,
     }
 
     try {
+      if (properties == null) {
```

Comment:
If properties are not set should it be treated as empty (i.e. empty map)?
If it's really invalid input, then we shouldn't returning INTERNAL_SERVER_ERROR. Maybe throw BadRequestException?
WDYT @albertshau ?

Reply:
It shouldn't be a 500 error. It could either be a 400 or a no-op, I'm leaning more towards a 400 (BadRequestException).

That being said, the jira linked here also should have a UI fix. If this API call fails, there should be some indication in the UI. The UI also shouldn't be making a request if there are no properties to put.

Reply:
also, this null check should happen outside of the try block and respond with a specific message about there not being any properties in the request. Catching NullPointerException doesn't guarantee properties are null. It can also happen if properties are non-null but writeArtifactProperties() threw a NPE for some reason.